crates/eth-rpc/src/servers/alchemy_rpc.rs

```diff
@@ -22,8 +30,117 @@ impl<P: Provider + Send + Sync> AlchemyRpc<P> {
 
 #[async_trait]
 impl<P: Provider + Send + Sync + 'static> AlchemyApiServer for AlchemyRpc<P> {
+    async fn token_allowance(
+        &self,
+        contract_address: Address,
+        account_address: Address,
+        spender_address: Address,
+    ) -> Result<TokenAllowance> {
+        let entry_point: Felt252Wrapper = keccak256("allowance(account, spender)")
```

Comment:
I wonder in general if we should cast 256-bit keccak into a 252 felt, with the possibility that the conversion may fail.
It'd be safer to pack it into two felts in case of failure, to not damage business logic, 
The packed two felts would then be inlined or destructured into calldata anyway. I may be wrong. 

This is a general question about the flow of the RPC and is pretty important, @ClementWalter may have an opinion

Reply:
I'll need guidance on this one, I'm not familiar with the tradeoffs between the two choices. I followed the `token_balances()` logic in the `core/src/client/mod.rs` file for this.

Reply:
I think this fixes it:
`FieldElement::from_byte_slice_be(&keccak256("balanceOf(address)").0[0..4])?;`

Reply:
Granted you implement 
```rust
impl<E: std::error::Error> From<FromByteSliceError> for EthApiError<E> {
    fn from(err: FromByteSliceError) -> Self {
        Self::ConversionError(format!("Failed to convert from byte slice: {}", err))
    }
}
```

Reply:
Are you running into the same error in that `impl doesn't use only types from inside the current crate`? I'm seeing an issue in that `FromByteSliceError` and `EthApiError` aren't defined in the current crate.

Reply:
Ahh sure! You'd have to put the above code snippet inside in crates>core>src>client>errors.rs

Reply:
That makes sense haha

Reply:
So the above code works in `token_balances` implementation in `mod.rs` but in the context of `alchemy_rpc.rs` I'm seeing that it is necessary to add `.map_err(EthApiError::<P::Error>::from)?;`

Reply:
Do we know why? Is it a problem of modules/scope in the sense that the conversion is not properly understood?

Reply:
Exactly, if I try to use the `?` operator without first using `map_err` then I'm met with the following error:

```rust
error[E0277]: `?` couldn't convert the error to `ErrorObject<'_>`
  --> crates/eth-rpc/src/servers/alchemy_rpc.rs:40:101
   |
40 |         let entry_point = FieldElement::from_byte_slice_be(&keccak256("balanceOf(address)").0[0..4])?;
   |                                                                                                     ^ the trait `From<FromByteSliceError>` is not implemented for `ErrorObject<'_>`
   |
   = note: the question mark operation (`?`) implicitly performs a conversion on the error value using the `From` trait
   = help: the following other types implement trait `From<T>`:
             <ErrorObject<'a> as From<jsonrpsee::jsonrpsee_types::error::ErrorCode>>
             <ErrorObject<'static> as From<EthApiError<E>>>
             <ErrorObject<'static> as From<ForkchoiceUpdateError>>
   = note: required for `std::result::Result<TokenAllowance, ErrorObject<'_>>` to implement `FromResidual<std::result::Result<std::convert::Infallible, FromByteSliceError>>`
```